system-parachains/coretime/coretime-kusama/src/migrations.rs

```diff
@@ -0,0 +1,232 @@
+				Ok(_) => log::info!(target: TARGET, "Request for 55 cores sent."),
+				Err(_) => log::error!(target: TARGET, "Request for 55 cores failed to send."),
+			}
+			match pallet_broker::Pallet::<Runtime>::start_sales(
+				RuntimeOrigin::root(),
+				5_000_000_000_0000,
+				55,
+			) {
+				Ok(_) => log::info!(target: TARGET, "Sales started"),
+				Err(_) => log::error!(target: TARGET, "Start sales failed!"),
+			}
+
+			// Weight for setting every lease and starting the sales, plus one read for leases
+			// check.
+			BrokerWeights::set_lease()
+				.saturating_mul(LEASES.len() as u64)
+				.saturating_add(BrokerWeights::request_core_count(55))
+				.saturating_add(BrokerWeights::start_sales(55))
+				.saturating_add(<Runtime as frame_system::Config>::DbWeight::get().reads(1))
+		}
+
+		#[cfg(feature = "try-runtime")]
+		fn pre_upgrade() -> Result<Vec<u8>, sp_runtime::TryRuntimeError> {
+			let sale_info = SaleInfo::<Runtime>::get().unwrap();
+			Ok(sale_info.encode())
+		}
+
+		#[cfg(feature = "try-runtime")]
+		fn post_upgrade(state: Vec<u8>) -> Result<(), TryRuntimeError> {
+			let prev_sale_info = <SaleInfoRecordOf<Runtime>>::decode(&mut &state[..]).unwrap();
+
+			// Idempotency hack - sorry. This just checks that the migration has run in the correct
+			// sale period, because the following checks only matter in that case. A bit of a "trust
+			// me bro", but this is the first sale for coretime-kusama.
+			if prev_sale_info.sale_start != 104193 {
+				log::info!(target: TARGET, "Idempotency hack has filtered checks for this run.");
+				return Ok(());
+			}
+
+			log::info!(target: TARGET, "Idempotency hack has not affected us for this run.");
+
+			let sale_info = SaleInfo::<Runtime>::get().unwrap();
+			let now = frame_system::Pallet::<Runtime>::block_number();
+			let config = Configuration::<Runtime>::get().unwrap();
+
+			// Check the sale start has changed as expected and the cores_offered is the correct
+			// number.
+			assert_eq!(sale_info.sale_start, now + config.interlude_length);
+			assert!(sale_info.region_begin > prev_sale_info.region_begin);
+			assert_eq!(sale_info.cores_offered, 3);
+
+			// The workplan entries start from the region begin reported by the new SaleInfo.
+			let workplan_start = sale_info.region_begin;
+
+			// Check the reservations are still in the workplan out of an abundance of caution.
+			for (core_id, task) in
+				[Task(1000), Task(1001), Task(1002), Task(1005), Pool].into_iter().enumerate()
+			{
+				assert_eq!(
+					Workplan::<Runtime>::get((workplan_start, core_id as u16)),
+					Some(Schedule::truncate_from(Vec::from([ScheduleItem {
+						mask: CoreMask::complete(),
+						assignment: task,
+					}])))
+				);
+			}
+
+			// Because we also run start_sales, 12 expiring leases are removed from the original 47,
+			// leaving 35.
+			let leases = Leases::<Runtime>::get();
+			assert_eq!(leases.len(), 35);
```

Comment:
```suggestion
			assert_eq!(leases.len(), LEASES.iter().filter(|(_, l)| sale_info.region_length * Timeslice::get() > l).count());
```

This should achieve the same?